core-plugins/src/main/java/co/cask/hydrator/plugin/batch/joiner/Joiner.java

```
@@ -0,0 +1,282 @@
+        Map<String, String> oldToAliases = stageToRenameField.getValue();
+        for (Map.Entry<String, String> entry : oldToAliases.entrySet()) {
+          if (Strings.isNullOrEmpty(entry.getKey()) || Strings.isNullOrEmpty(entry.getValue())) {
+            throw new IllegalArgumentException("Either field name or alias is missing.");
+          }
+          perStageFieldsToRename.put(stageToRenameField.getKey(), entry.getKey(), entry.getValue());
+        }
+      }
+    } else {
+      throw new IllegalArgumentException("selectedFields can not be empty. Please provide at least 1 selectedFields");
+    }
+    return perStageFieldsToRename;
+  }
+
+  Iterable<String> getRequiredInputs() {
+    if (conf.getRequiredInputs() != null) {
+      return Splitter.on(',').trimResults().omitEmptyStrings().split(conf.getRequiredInputs());
+    }
+    return Collections.<String>emptyList();
+  }
+
+  @Path("outputSchema")
+  public Schema getOutputSchema(GetSchemaRequest request) {
+    try {
+      return getOutputSchema(request.inputSchemas);
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException(e.getMessage());
+    }
+  }
+
+  /**
+   * Endpoint request for output schema.
+   */
+  public static class GetSchemaRequest extends JoinerConfig {
+    public Map<String, Schema> inputSchemas;
+  }
+
+
+  private Schema getOutputSchema(Map<String, Schema> inputSchemas) {
+    validateRequiredInputs(inputSchemas);
+    // sort the input schemas by input names to get the deterministic order of fields for output schema
+    Map<String, Schema> inputs = new TreeMap<>(inputSchemas);
+    Map<String, Schema.Field> outputFields = new LinkedHashMap<>();
+    Map<String, String> duplicateFieldsMap = new HashMap<>();
+
+    for (Map.Entry<String, Schema> input : inputs.entrySet()) {
+      String stageName = input.getKey();
+      Schema inputSchema = input.getValue();
```

You need to handle the case where this schema is null. For now you could throw an exception about all inputs need to have a schema set, then add a TODO and jira to handle cases where schema is not known at configure time.